lib/smartsheet/util/atomize_keys.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Smartsheet.Util.AtomizeKeys do
+  def atomize_keys(nil), do: nil
+
+  def atomize_keys(struct = %{__struct__: _}) do
+    struct
+  end
+
+  def atomize_keys(map = %{}) do
+    map
+    |> Enum.map(fn
+      {k, v} when is_atom(k) -> {k, atomize_keys(v)}
+      {k, v} -> {String.to_atom(k), atomize_keys(v)}
+    end)
+    |> Enum.into(%{})
+  end
+
+  def atomize_keys([head | rest]) do
+    [atomize_keys(head) | atomize_keys(rest)]
```

Comment:
Don't think you should change anything but just an FYI on atoms in Elixir, they are never garbage collected. So generally you want to be super careful about converting user supplied params to atoms because you can't run out of VM memory space.